src/components/CardView.tsx

```diff
@@ -1,114 +1,44 @@
+import { useEffect, useState } from 'react';
 import { Card } from '../types/card';
+import CardFrontView from './CardFrontView';
+import CardBackView from './CardBackView';
 import styled from 'styled-components';
-import CardChip from '../assets/image/cardChip.png';
-import Visa from '../assets/image/Visa.png';
-import Master from '../assets/image/Mastercard.png';
-import { SECRET_NUMBER, SLASH } from '../constants/system';
-import { isMasterCard, isVisaCard } from '../utils/checkCardType';
 
 interface Props {
   cardInfo: Card;
 }
 
 export default function CardView({ cardInfo }: Props) {
-  const { cardNumbers, expiryDate, userName } = cardInfo;
+  const [showBack, setShowBack] = useState(false);
 
-  const getCardType = (cardNumber: string) => {
-    const cardBrandNumber = parseInt(cardNumber.substring(0, 2), 10);
-
-    if (isVisaCard(cardBrandNumber)) return 'visa';
-    if (isMasterCard(cardBrandNumber)) return 'master';
-  };
-
-  const getCardImage = (cardNumber: string) => {
-    const cardType = getCardType(cardNumber);
-
-    if (cardType === 'visa') return Visa;
-    if (cardType === 'master') return Master;
-  };
-
-  const cardImgSrc = getCardImage(cardNumbers.cardNumber1);
-
-  const formatDate = (date: string) => {
-    if (date.length === 1) {
-      return `0${date}`;
-    }
-    return date;
-  };
+  useEffect(() => {
+    if (cardInfo.cvc) setShowBack(true);
+    if (cardInfo.cvc.length === 3) setShowBack(false);
+  }, [cardInfo.cvc]);
 
   return (
-    <CardContainer>
-      <ImgBox>
-        <CardImg src={CardChip} />
-        {cardImgSrc && <CardImg src={cardImgSrc} />}
-      </ImgBox>
-
-      <CardNumbers>
-        <CardNumber>{cardNumbers.cardNumber1}</CardNumber>
-        <CardNumber>{cardNumbers.cardNumber2}</CardNumber>
-        <SecretCardNumber>
-          {SECRET_NUMBER.repeat(cardNumbers.cardNumber3.length)}
-        </SecretCardNumber>
-        <SecretCardNumber>
-          {SECRET_NUMBER.repeat(cardNumbers.cardNumber4.length)}
-        </SecretCardNumber>
-      </CardNumbers>
-      <TextBox>
-        {formatDate(expiryDate.month)}
-        {expiryDate.year.length > 0 ? SLASH : ''}
-        {formatDate(expiryDate.year)}
-      </TextBox>
-      <TextBox>{userName}</TextBox>
-    </CardContainer>
+    <CardsContainer showBack={showBack}>
+      <Front> {!showBack && <CardFrontView cardInfo={cardInfo} />}</Front>
+      <Back> {showBack && <CardBackView cvc={cardInfo.cvc} />}</Back>
```

Comment:
```suggestion
      {showBack 
        ? <Back><CardBackView cvc={cardInfo.cvc} /></Back>
        : <Front><CardFrontView cardInfo={cardInfo} /></Front>
      }
```

삼항 연산자로 처리할 수도 있을 것 같은데요,
어떤 게 더 낫다고 보시나요?? (삼항이 맞다거나 더 낫다는 것은 아니고 의견이 궁금합니다!)

Reply:
삼항 연산자가 state 하나의 True/False 에 대해 다른 컴포넌트를 가져온다는 코드를 이해하는데 더 쉬울거 같네요! 반영하겠습니다!